### PR TITLE
Add ThermoPro TP28b

### LINKED ISSUE #2877: Support for Thermopro TP-28B
My best guess from the following is, `rtl_433 -f 915M -X "n=name,m=FSK_PCM,s=100,l=100,r=4096"`. The `r=4096` is a just something I put to make the error go away. I have no idea how to figure out `reset`.

```
rtl_433 -f 915M -A

Detected FSK package	2024-03-14 12:41:00
Analyzing pulses...
Total count:   90,  width: 51.69 ms		(51688 S)
Pulse width distribution:
 [ 0] count:    1,  width:    0 us [0;0]	(   0 S)
 [ 1] count:    5,  width:    2 us [2;2]	(   2 S)
 [ 2] count:    2,  width:    8 us [8;9]	(   8 S)
 [ 3] count:    2,  width:    5 us [5;6]	(   5 S)
 [ 4] count:    1,  width:    3 us [3;3]	(   3 S)
 [ 5] count:    3,  width:    1 us [1;1]	(   1 S)
 [ 6] count:   61,  width:  105 us [104;110]	( 105 S)
 [ 7] count:   12,  width:  212 us [210;215]	( 212 S)
 [ 8] count:    3,  width:  319 us [319;320]	( 319 S)
Gap width distribution:
 [ 0] count:    1,  width:    2 us [2;2]	(   2 S)
 [ 1] count:    5,  width:    3 us [3;3]	(   3 S)
 [ 2] count:    2,  width:    1 us [1;1]	(   1 S)
 [ 3] count:    2,  width:    5 us [5;5]	(   5 S)
 [ 4] count:   55,  width:  107 us [105;112]	( 107 S)
 [ 5] count:    8,  width:  214 us [214;216]	( 214 S)
 [ 6] count:    3,  width:  320 us [320;321]	( 320 S)
 [ 7] count:    6,  width:  504 us [429;536]	( 504 S)
 [ 8] count:    2,  width: 1922 us [1817;2028]	(1922 S)
 [ 9] count:    2,  width:  642 us [640;644]	( 642 S)
 [10] count:    1,  width: 4881 us [4881;4881]	(4881 S)
 [11] count:    1,  width: 18733 us [18733;18733]	(18733 S)
 [12] count:    1,  width: 1267 us [1267;1267]	(1267 S)
Pulse period distribution:
 [ 0] count:    1,  width:    2 us [2;2]	(   2 S)
 [ 1] count:    4,  width:    5 us [5;6]	(   5 S)
 [ 2] count:    2,  width:    9 us [9;9]	(   9 S)
 [ 3] count:    1,  width:   14 us [14;14]	(  14 S)
 [ 4] count:    1,  width:    7 us [7;7]	(   7 S)
 [ 5] count:    1,  width:    3 us [3;3]	(   3 S)
 [ 6] count:    1,  width:  109 us [109;109]	( 109 S)
 [ 7] count:   49,  width:  213 us [209;217]	( 213 S)
 [ 8] count:    9,  width:  320 us [317;325]	( 320 S)
 [ 9] count:    8,  width:  430 us [424;461]	( 430 S)
 [10] count:    3,  width:  641 us [639;644]	( 641 S)
 [11] count:    4,  width:  854 us [854;855]	( 854 S)
 [12] count:    2,  width: 2135 us [2028;2243]	(2135 S)
 [13] count:    1,  width: 4987 us [4987;4987]	(4987 S)
 [14] count:    1,  width: 18839 us [18839;18839]	(18839 S)
 [15] count:    1,  width: 1268 us [1268;1268]	(1268 S)
Pulse timing distribution:
 [ 0] count:    1,  width:    0 us [0;0]	(   0 S)
 [ 1] count:    7,  width:    2 us [2;2]	(   2 S)
 [ 2] count:    2,  width:    8 us [8;9]	(   8 S)
 [ 3] count:    4,  width:    5 us [5;6]	(   5 S)
 [ 4] count:    6,  width:    3 us [3;3]	(   3 S)
 [ 5] count:    5,  width:    1 us [1;1]	(   1 S)
 [ 6] count:  116,  width:  106 us [104;112]	( 106 S)
 [ 7] count:   20,  width:  213 us [210;216]	( 213 S)
 [ 8] count:    6,  width:  319 us [319;321]	( 319 S)
 [ 9] count:    6,  width:  504 us [429;536]	( 504 S)
 [10] count:    2,  width: 1922 us [1817;2028]	(1922 S)
 [11] count:    2,  width:  642 us [640;644]	( 642 S)
 [12] count:    1,  width: 4881 us [4881;4881]	(4881 S)
 [13] count:    1,  width: 18733 us [18733;18733]	(18733 S)
 [14] count:    1,  width: 1267 us [1267;1267]	(1267 S)
Level estimates [high, low]:   2565,      5
RSSI: -8.1 dB SNR: 27.1 dB Noise: -35.2 dB
Frequency offsets [F1, F2]:   -3151,   3303	(-48.1 kHz, +50.4 kHz)
Guessing modulation: Non Return to Zero coding (Pulse Code)
Attempting demodulation... short_width: 1, long_width: 1, reset_limit: 1024, sync_width: 0
Use a flex decoder with -X 'n=name,m=FSK_PCM,s=1,l=1,r=1024'
```

### PR DESCRIPTION
**Pull request by joshpearce**
resolve #2877

### CODE REVIEW

File: src/devices/thermopro_tp28b.c [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        210 + 2.8 = 212.8 C (displayed rounded to 213)
+
+    - p1_temp: probe 1 current temp. 16 bit BCD
+    - p1_set_hi: probe 1 high alarm temp. 16 bit BCD
+    - p1_set_lo: probe 1 low alarm temp. 16 bit BCD
+    - p2_temp: probe 2 current temp. 16 bit BCD
+    - p2_set_hi: probe 2 high alarm temp. 16 bit BCD
+    - p2_set_lo: probe 2 low alarm temp. 16 bit BCD
+
+    Below is some work on status/alarm flags, but I can't quite make sense of them all.
```
Comment (zuckschwerdt) [inline src/devices/thermopro_tp28b.c:49]: Normal text and lists shouldn't be indented

File: src/devices/thermopro_tp28b.c [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+
+#include "decoder.h"
+
+/**
```
Comment (zuckschwerdt) [inline src/devices/thermopro_tp28b.c:14]: When using helper functions this line needs to be:
`/** @fn int thermopro_tx2c_decode(r_device *decoder, bitbuffer_t *bitbuffer)`

File: src/devices/thermopro_tp28b.c [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    float p1_temp   = bcd2float(b[0], b[1]);
+    float p1_set_hi = bcd2float(b[2], b[3]);
+    float p1_set_lo = bcd2float(b[4], b[5]);
+    float p2_temp   = bcd2float(b[6], b[7]);
+    float p2_set_hi = bcd2float(b[8], b[9]);
+    float p2_set_lo = bcd2float(b[10], b[11]);
+
+    /* clang-format off */
+    data_t *data = data_make(
+            "model",                "",                             DATA_STRING,    "TP28b",
```
Comment (zuckschwerdt) [inline src/devices/thermopro_tp28b.c:133]: "ThermoPro-TP28b" would be a better model key

**Comment by zuckschwerdt** [comment]
Looks really good to me.

File: src/devices/thermopro_tp28b.c (hunk 2 above)
Comment (joshpearce) [reply src/devices/thermopro_tp28b.c]: Thanks. And I changed the function name 😄

**Pull request merged**